firefox-ios/Client/Utils/SponsoredContentFilterUtility.swift

```diff
@@ -15,7 +15,15 @@ struct SponsoredContentFilterUtility {
     }
 
     func filterSponsoredSites(from sites: [Site]) -> [Site] {
-        return sites.filter { !$0.url.contains(hideWithSearchParam) }
+        return sites.filter { item in
+            guard let url = URL(string: item.url) else {return true}
+
+            return !(containsSearchParam(url: url) ?? false)
+        }
+    }
+
+    func containsSearchParam(url: URL) -> Bool? {
+        return (url.query?.split(separator: "&").map({ String($0) }).contains(hideWithSearchParam))
     }
```

Comment:
```suggestion
    func containsSearchParam(url: URL) -> Bool {
        return url.query?.split(separator: "&").contains { $0 == hideWithSearchParam } ?? false
    }
```

Oops, I forgot that `split` returns `[Substring]`, not `[String]`, so we can't use `.contains(hideWithSearchParam)` directly on the result. But we can compare a `Substring` and a `String` with `==`, so we can avoid the extra `.map` and conversion.

Also, let's not use `Bool?` as a return type. Optional Booleans can be a bit of a code smell, because it's not always clear how or why `false` differs from `nil`. I have a preference for either using an explicit `enum` (when you need a tri-state), or a non-optional Boolean (if `false` and `nil` mean the same thing).

In this case, if the URL doesn't have a query, we know it can't contain the search param, so let's return `false` here. That simplifies the other call sites, too ☺️